src/javaRosa.js

```diff
@@ -1814,6 +1814,10 @@ define([
                             }
                             writeValue(xmlWriter, val);
                             xmlWriter.writeEndElement();
+                            if (val.search(/#case\//) !== -1) {
+                                form_.addInstanceIfNotExists({ id: 'commcaresession' });
+                                form_.addInstanceIfNotExists({ id: 'casedb' });
+                            }
```

Comment:
What if, instead of on write XML, we did something similar on itext change? The down-side is that simply loading and saving a form will not "fix" (add instances for) case references in itext. I'm not excited about moving the instances after the `<itext>` node either, but my excitement level aside, we should check if that's even supported if we really think that's going to happen.